src/cli.ts

```diff
@@ -118,29 +119,37 @@ function debugCliInfo(version: string, channel: string, env: Env, config: IConfi
   );
 }
 
-interface VersionDetail {
+export interface VersionDetail {
   cliVersion: string;
   architecture: string;
   nodeVersion: string;
   pluginVersions?: string[];
   osVersion?: string;
+  shell?: string;
+  binLocation?: string;
+}
+
+// Function which returns Version object which includes cli version, plugin versions, OS and its version.
+function getVersionDetail(isVerbose: boolean, config: IConfig): VersionDetail {
+  const versions = config.userAgent.split(' ');
+  const cliVersion: string = versions[0];
+  const architecture: string = versions[1];
+  const nodeVersion: string = versions[2];
+  if (!isVerbose) return { cliVersion, architecture, nodeVersion };
+  const pluginVersion: string = exec('sfdx plugins --core', {
```

Comment:
can we get this info from the oclif config, or `plugin-plugins`? It might be faster than `exec`ing out for it

Reply:
Yeah, we definitely should not be shelling out for this.
See [this](https://github.com/oclif/plugin-version/blob/65f6220479311714ed775cf6647cd2931f7b5bc7/src/commands/version.ts#L32-L36) as an example of how to get the plugins without running `sfdx plugins --core`.

Reply:
I'll look into it in the next PR.  I think you're right.